In [1]:
#Importing needed libraries
import pandas as pd
import plotly.express as px
import streamlit as st

In [2]:
#reading data into a dataframe
df_vehicles_us = pd.read_csv('/Users/octaviodon/Documents/GitHub/tripleten_sprint_7_proyecto/vehicles_us.csv')

In [3]:
#imprimimos la info y head del df_vehicles_us

df_vehicles_us.info(show_counts=True)
print('--' * 20)
print()
df_vehicles_us_head = df_vehicles_us.head(10)
df_vehicles_us_head

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51525 entries, 0 to 51524
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   price         51525 non-null  int64  
 1   model_year    47906 non-null  float64
 2   model         51525 non-null  object 
 3   condition     51525 non-null  object 
 4   cylinders     46265 non-null  float64
 5   fuel          51525 non-null  object 
 6   odometer      43633 non-null  float64
 7   transmission  51525 non-null  object 
 8   type          51525 non-null  object 
 9   paint_color   42258 non-null  object 
 10  is_4wd        25572 non-null  float64
 11  date_posted   51525 non-null  object 
 12  days_listed   51525 non-null  int64  
dtypes: float64(4), int64(2), object(7)
memory usage: 5.1+ MB
----------------------------------------



price  model_year           model  condition  cylinders fuel  odometer  \
0   9400      2011.0          bmw x5       good        6.0  gas  145000.0   
1  25500         NaN      ford f-150       good        6.0  gas   88705.0   
2   5500      2013.0  hyundai sonata   like new        4.0  gas  110000.0   
3   1500      2003.0      ford f-150       fair        8.0  gas       NaN   
4  14900      2017.0    chrysler 200  excellent        4.0  gas   80903.0   
5  14990      2014.0    chrysler 300  excellent        6.0  gas   57954.0   
6  12990      2015.0    toyota camry  excellent        4.0  gas   79212.0   
7  15990      2013.0     honda pilot  excellent        6.0  gas  109473.0   
8  11500      2012.0     kia sorento  excellent        4.0  gas  104174.0   
9   9200      2008.0     honda pilot  excellent        NaN  gas  147191.0   

  transmission    type paint_color  is_4wd date_posted  days_listed  
0    automatic     SUV         NaN     1.0  2018-06-23           19  
1    automatic  pickup       white     1.0  2018-10-19           50  
2    automatic   sedan         red     NaN  2019-02-07           79  
3    automatic  pickup         NaN     NaN  2019-03-22            9  
4    automatic   sedan       black     NaN  2019-04-02           28  
5    automatic   sedan       black     1.0  2018-06-20           15  
6    automatic   sedan       white     NaN  2018-12-27           73  
7    automatic     SUV       black     1.0  2019-01-07           68  
8    automatic     SUV         NaN     1.0  2018-07-16           19  
9    automatic     SUV        blue     1.0  2019-02-15           17

In [4]:
#Veryfying duplicates
df_vehicles_us.duplicated().sum()

np.int64(0)

In [5]:
#veryfing missing values
df_vehicles_us.isnull().sum()

price               0
model_year       3619
model               0
condition           0
cylinders        5260
fuel                0
odometer         7892
transmission        0
type                0
paint_color      9267
is_4wd          25953
date_posted         0
days_listed         0
dtype: int64

### Findings

- Columns   
    - model_year 
        - fill with median by model - 3619
        - type float64 --> int64
    - cylinders
        - null values -5260 into "unknown"
    - odometer
        - fill nulls with median by model+year
        - type float64 --> int64
    - paint_color
        - fill nulls with "unknown"
    - is_4wd
        - Only has 1's and NaN, nulls will be replaced with 0 to properly analyze.

In [ ]:
#replacing missing values in 'model_year' with the median
median_model_year = df_vehicles_us['model_year'].median()
#df_vehicles_us['model_year'].fillna(median_model_year, inplace=True) #old version
df_vehicles_us.fillna({'model_year' : median_model_year},inplace=True)

#changing model_year to integer
df_vehicles_us['model_year'] = df_vehicles_us['model_year'].astype('int64')

#replacing missing values in "cylinders" with "unknown"
df_vehicles_us.fillna({'cylinders': 'unknown'}, inplace=True)
#converting odometer to integer with NA handling
df_vehicles_us['odometer'] = df_vehicles_us['odometer'].astype('Int64') 
#replacing odometer missing values with the median
median_odometer = df_vehicles_us['odometer'].median()
df_vehicles_us.fillna({'odometer': median_odometer}, inplace=True)
#changing odometer to integer
df_vehicles_us['odometer'] = df_vehicles_us['odometer'].astype('int64') 

#replacing paint_color missing values with "unknown"
df_vehicles_us.fillna({'paint_color': 'unknown'}, inplace=True)
#replacing is_4wd missing values with 0
df_vehicles_us.fillna({'is_4wd': 0}, inplace=True) 
#changing is_4wd to integer
df_vehicles_us['is_4wd'] = df_vehicles_us['is_4wd'].astype('int64')

In [11]:
#showinf info after cleaning
df_vehicles_us.info(show_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51525 entries, 0 to 51524
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   price         51525 non-null  int64  
 1   model_year    51525 non-null  int64  
 2   model         51525 non-null  object 
 3   condition     51525 non-null  object 
 4   cylinders     51525 non-null  object 
 5   fuel          51525 non-null  object 
 6   odometer      51525 non-null  int64  
 7   transmission  51525 non-null  object 
 8   type          51525 non-null  object 
 9   paint_color   51525 non-null  object 
 10  is_4wd        51525 non-null  float64
 11  date_posted   51525 non-null  object 
 12  days_listed   51525 non-null  int64  
dtypes: float64(1), int64(4), object(8)
memory usage: 5.1+ MB
